### DISCUSSION TITLE
Any recommendations between DoH, DoT, DoQ for upstreams?

### DISCUSSION THREAD

**Discussion by timkgh** [AdguardTeam/AdGuardHome#3614] [Q&A]
If an upstream supports all DoX protocols and there are no restrictions such as firewalls, any recommendations for the protocol to use for max performance? E.g. does DoT keep connections alive? Is DoQ significantly faster?

**Comment by ameshkov** [comment]
All three keep connections alive. Depends on your network connection. If it's stable then all three are interchangeable. If it's not stable,  DoQ should be preferred.

**Comment by timkgh** [comment]
@ameshkov There may be a problem with DoT, I observe poor performance with it:
https://github.com/AdguardTeam/AdGuardHome/issues/3647
